Reduce the frequency of refreshing connections

hewison-chris opened the pull request:
By using the `network_discovery_interval` duration which is usually set
to a larger duration than the `retry_delay`.

hewison-chris committed:
Reduce the frequency of refreshing peer addresses

By using the `network_discovery_interval` duration which is usually set
to a larger duration than the `retry_delay`. Also exit gracefully when
shutting down.

omerfirmak commented on source/agora/network/Manager.d:
```diff
@@ -343,7 +343,7 @@ public class NetworkManager
             {
                 scope (success)
-                    this.outer.taskman.wait(this.outer.node_config.retry_delay);
+                    this.outer.taskman.wait(this.outer.node_config.network_discovery_interval);
```
I feel like the right aproach would be something like;

```
while (1)
{
   foreach (client; clients)
   {
      // work
     wait(retry_delay); // totally optional
   }
   wait(network_discovery_interval);
}
```

Otherwise it would take a really long time to go thru all the clients

hewison-chris replied:
Updated

omerfirmak commented on source/agora/network/Manager.d:
```diff
@@ -340,32 +340,48 @@ public class NetworkManager
-                catch (Exception ex)
+                auto peers = this.clients[].walkLength;
+                auto i = 1;
+                while (this.is_running && i <= peers)
```
Let's make this a for loop to get rid of the temp variable?

`foreach (_; 0..this.clients[].walkLength)` ?

hewison-chris replied:
Updated

hewison-chris committed:
NetworkDiscovery Unit test: Remove workaround

The test can now expect the count of addresses to be exactly the number
of nodes plus the registry. Previously the node would sometimes connect
to itself but now it always bans itself but the address is always
included in node info.

hewison-chris committed:
GossipProtocol unit test: Correct network test config

omerfirmak reviewed (approved):
Merge at will
